### ISSUE TITLE
Crash in `MaterialXCore::Element::setChildIndex()` due to off-by-one error in index check

### ISSUE THREAD

**Issue by StefanHabel** [AcademySoftwareFoundation/MaterialX#1581]
I believe I found an off-by-one crash bug in [`MaterialXCore::Element::setChildIndex()`](https://github.com/AcademySoftwareFoundation/MaterialX/blob/main/source/MaterialXCore/Element.cpp#L162-L178).

Minimal repro via Python bindings:
```bash
$ python3 -c 'import MaterialX as mx; d = mx.createDocument(); b = d.addBackdrop(); d.setChildIndex(b.getName(), 1)'
Segmentation fault
```
Expected behavior:
```python
Traceback (most recent call last):
  File "<string>", line 1, in <module>
LookupError: Invalid child index
```
Fix:
```diff
-    if (index < 0 || index > (int) _childOrder.size())
+    if (index < 0 || index >= (int) _childOrder.size())
     {
         throw Exception("Invalid child index");
     }
```

Found while working on https://github.com/AcademySoftwareFoundation/MaterialX/issues/342

**Comment by jstone-lucasfilm** [comment]
This looks like a good catch, thanks @StefanHabel, and feel free to post your fix if you have the bandwidth!

**Comment by jstone-lucasfilm** [comment]
Thanks for this report, @StefanHabel, as well as for the fix in #1582!

**Comment by StefanHabel** [comment]
Thanks very much @jstone-lucasfilm ! 🙌
